netbox/operations.py

```diff
@@ -0,0 +1,428 @@
+        raise ConnectorError(str(err))
+
+
+def get_device(config: dict, params: dict):
+    try:
+        nb = Netbox(config)
+        params = _build_payload(params)
+
+        endpoint = f'/api/dcim/devices/{params.get("id")}'
+        return nb.make_request(endpoint=endpoint, method='GET', params=params)
+    except Exception as err:
+        logger.error(str(err))
+        raise ConnectorError(str(err))
+
+
+def update_device(config: dict, params: dict):
+    try:
+        nb = Netbox(config)
+        params = _build_payload(params)
+
+        endpoint = f'/api/dcim/devices/{params.get("id")}'
+        return nb.make_request(endpoint=endpoint, method='PATCH', data=params.get("patch_fields"))
+    except Exception as err:
+        logger.error(str(err))
+        raise ConnectorError(str(err))
+
+
+def delete_device(config: dict, params: dict):
+    try:
+        nb = Netbox(config)
+        params = _build_payload(params)
+
+        endpoint = f'/api/dcim/devices/{params.get("id")}'
+        return nb.make_request(endpoint=endpoint, method='DELETE')
+    except Exception as err:
+        logger.error(str(err))
+        raise ConnectorError(str(err))
+
+
+def get_cable_list(config: dict, params: dict):
+    try:
+        nb = Netbox(config)
+        params = _build_payload(params)
+
+        if params.get("other_fields") is not None:
+            other_fields = params.pop("other_fields")
+            params.update(other_fields)
+
+        endpoint = '/api/dcim/cables/'
+        return nb.make_request(endpoint=endpoint, method='GET', params=params)
+    except Exception as err:
+        logger.error(str(err))
+        raise ConnectorError(str(err))
+
+
+def get_cable(config: dict, params: dict):
+    try:
+        nb = Netbox(config)
+        params = _build_payload(params)
+
+        endpoint = f'/api/dcim/cables/{params.get("id")}'
+        return nb.make_request(endpoint=endpoint, method='GET', params=params)
+    except Exception as err:
+        logger.error(str(err))
+        raise ConnectorError(str(err))
+
+
+def update_cable(config: dict, params: dict):
+    try:
+        nb = Netbox(config)
+        params = _build_payload(params)
+
+        endpoint = f'/api/dcim/cables/{params.get("id")}'
+        return nb.make_request(endpoint=endpoint, method='PATCH', data=params.get("patch_fields"))
+    except Exception as err:
+        logger.error(str(err))
+        raise ConnectorError(str(err))
+
+
+def delete_cable(config: dict, params: dict):
+    try:
+        nb = Netbox(config)
+        params = _build_payload(params)
+
+        endpoint = f'/api/dcim/cables/{params.get("id")}'
+        return nb.make_request(endpoint=endpoint, method='DELETE')
+    except Exception as err:
+        logger.error(str(err))
+        raise ConnectorError(str(err))
+
+
+def _check_health(config):
+    try:
+        tg = Netbox(config)
+        endpoint = '/api/dcim/sites/'
+        response = tg.make_request(endpoint=endpoint)
```

Comment:
if possible provide limit parameter . No need to fetch more data for health check